cloud_testing/xctest_parser/src/main/kotlin/xctest/Shard.kt

```diff
@@ -0,0 +1,81 @@
+package xctest
+
+import com.dd.plist.NSArray
+import com.dd.plist.NSDictionary
+import com.dd.plist.PropertyListParser
+import java.io.File
+import java.nio.file.Paths
+
+object Shard {
+
+    private fun testsForTarget(testDictionary: NSDictionary, testTarget: String, testRoot: String): Set<String> {
+        val productPaths = (testDictionary["DependentProductPaths"] as NSArray)
+        for (product in productPaths.array) {
+            val productString = product.toString()
+            if (productString.contains("/$testTarget.xctest")) {
+                val binaryRoot = productString.replace("__TESTROOT__/", testRoot)
+                println("Found product: $binaryRoot")
+
+                val isSwift = Paths.get(binaryRoot, "libswiftRemoteMirror.dylib").toFile().exists()
+                val binaryName = File(binaryRoot).nameWithoutExtension
+                val binaryPath = Paths.get(binaryRoot, binaryName).toString()
+
+                val methods = if (isSwift) {
+                    println("Swift detected")
+                    Parse.parseSwiftTests(binaryPath)
+                } else {
+                    println("Objc detected")
+                    Parse.parseObjcTests(binaryPath)
+                }
```

Comment:
can't `methods` be a mixture of swift + objectiveC?

Reply:
That's an interesting question. I believe that when writing UI automation, teams will pick one language. I'm not sure Xcode supports having a testing target with tests written in both Swift and Objective C. The teams I've talked with are writing EarlGrey tests in ObjC or Swift, but not both.

Reply:
Sure, using a single language makes a lot of sense but AFAIK firebase doesn't differentiate them. why wouldn't methods be:

```
  methods = Parse.parseObjcTests(binaryPath) + Parse.parseSwiftTests(binaryPath)
```

Reply:
Xcode does allow test targets to contain both Objective C and Swift tests. Xcode can also run all the tests in as a single test run. 

We could modify `Parse` to be language agnostic and run aggregate the results of both parsing methods. That would also allow end users to just specify a binary and not have to declare whether it is objc, swift, or mixed.

Reply:
I wonder if that's what bluepill does.

Reply:
looks like that's the right call, bluepill is running both and concatenating the results.

https://github.com/linkedin/bluepill/blob/f45878114e2b32899ca121c971adb0f638529d94/Source/Shared/BPXCTestFile.m#L56

Reply:
Fixed.